### PR TITLE
Clear has_info_request on whiteboard and developer_comments (bug 1117329)

### PR DESCRIPTION
**Pull request by mstriemer**
https://bugzilla.mozilla.org/show_bug.cgi?id=1117329

This clears the `has_info_request` flag when the addon's `whiteboard` or `developer_comments` are updated`. Still need to check on new version upload and source package upload but those might be working already.

### CODE REVIEW

File: apps/addons/tests/test_models.py [hunk 1]
```diff
@@ -2294,6 +2294,65 @@ def test_enable_addon(self, file_mock):
...
+        addon.versions.create(has_info_request=True)
+        return addon
+
+    def assertHasInfoSet(self, addon):
+        assert any([v.has_info_request for v in addon.versions.all()])
+
+    def assertHasInfoNotSet(self, addon):
+        assert all([not v.has_info_request for v in addon.versions.all()])
+
+    def test_has_info_save(self):
```
Comment (magopian) [inline apps/addons/tests/test_models.py]: Docstring please (something like `Saving an add-on without changing whiteboard or adding comments doesn't change the has_info_request`)

File: apps/addons/tests/test_models.py [hunk 2]
```diff
@@ -2294,6 +2294,65 @@ def test_enable_addon(self, file_mock):
...
+        addon = self.make_addon()
+        self.assertHasInfoSet(addon)
+        addon.save()
+        self.assertHasInfoSet(addon)
+
+    def test_has_info_update_whiteboard(self):
```
Comment (magopian) [inline apps/addons/tests/test_models.py]: Same here, docstring explaining exactly what the test is testing. The test name does help, but I'd rather help our future selves remember exactly what this is about ;)
(Same for the tests below)

**Comment by magopian** [comment]
This is nice, I like those clear and concise tests!
r+wc

**Comment by mstriemer** [comment]
I added docstrings to all the tests and also made it so that clearing `whiteboard` or `developer_comments` doesn't clear the `has_info_request` flag (with tests).

**Comment by magopian** [comment]
You have a flake8 error, and the tests naming isn't very consistent (eg `test_has_info_update_whiteboard_removed` vs `test_has_info_remove_developer_comments`) but that's just nitpicking ;)

r+wc

**Comment by magopian** [comment]
That's excellent, thanks @mstriemer, r+

**Pull request merged**